### ISSUE TITLE
istanbul chainId opcode returns bad value

### ISSUE THREAD

**Issue by Amxx** [trufflesuite/ganache-cli#700]
**Steps to reproduce**

1. Start ganache-cli using the following command:

```
ganache-cli -l 8000000 -i 65535 -h 0.0.0.0 -k istanbul
```

2. Compile/deploy the following contract:

```
contract ChainIdTest
{
	function chainId()
	public pure returns (uint256 id)
	{
		assembly { id := chainid() }
	}
}
```

3. Call the `chainId()` function.

I expect the returned value to be the chainId passed using the `-i` flag and accessible using `await web3.eth.net.getId();` ... but I get a value of `1`

**Comment by Amxx** [comment]
After some investigation, this is what I've noticed:

- The chainId is retreived from the `ethereumjs-common` object used by `ethereumjs-vm` (a dependency of ganache-core):
https://github.com/ethereumjs/ethereumjs-vm/blob/master/lib/evm/eei.ts#L288

- In our case, this `common` object is created by the constructor of the ethereumjs-vm. This uses a `chain` field from the opts (or mainnet by default):
https://github.com/ethereumjs/ethereumjs-vm/blob/master/lib/index.ts#L103

- Apparently, ganache-core does not pass any `chain` field when constructing the VM
https://github.com/trufflesuite/ganache-core/blob/develop/lib/blockchain_double.js#L128

- This is why the "mainnet" is selected by default, and why I get `1` when doing my test.

- A "obvious" solution would be to pass the chainId to the VM, which will then be interpreted by the `Common` constructor:
https://github.com/ethereumjs/ethereumjs-common/blob/master/src/index.ts

- If we put a number into the chain field, common pass it from the constuctor to `setChain` then to `_getChainParams` which will then lookup the name of the of this chain in `chainParams['names']`
https://github.com/ethereumjs/ethereumjs-common/blob/master/src/index.ts#L50

- chainParams only knows mainnet, ropsten, rinkeby an goerli (strangly with number 6284 and not 5)
https://github.com/ethereumjs/ethereumjs-common/blob/master/src/chains/index.ts

Long story short, fixing this needs:
- Passing the chainid to ethereumjs-common

AND

- fixing ethereumjs-common to accept arbitrary chain ids
